linchpin/cli/__init__.py

```diff
@@ -138,7 +138,18 @@ def _write_to_inventory(self, tx_id=None, inv_path=None, inv_format="cfg"):
                         lt_data = targets[name]["inputs"]["layout_data"]
                         t_data = targets[name]["inputs"]["topology_data"]
                         layout = lt_data["inventory_layout"]
-                        i_path = targets[name]["outputs"]["inventory_path"][0]
+                        # check whether inventory_file is mentioned in layout
+                        if layout.get("inventory_file", None):
+                            i_path = layout["inventory_file"]
+                            # if its not absolute path make path
+                            # relative to workspace
+                            if not os.path.isabs(i_path):
+                                i_path = "{0}/{1}".format(self.ctx.workspace,
+                                                          i_path)
+                        else:
+                            i_path = targets[name]["outputs"]
+                            i_path = i_path["inventory_path"][0]
```

Comment:
Is this correct?  it looks to me like you meant to build upon the previous i_path definition.

Reply:
@p3ck : Flake 8 tests were failing due to the length of line was exceeding 80 characters when I tried to accomodate everything into single line. I had to break it down into two lines as above to get the flake8 test pass.

Reply:
thanks for the explanation